api/v1alpha1/release_types.go

```diff
@@ -158,19 +168,28 @@ func (r *Release) IsDone() bool {
 	return condition != nil && condition.Status != metav1.ConditionUnknown
 }
 
-// MarkDeployed sets the AllComponentsDeployed status in the Release to True with the provided reason and message.
-func (r *Release) MarkDeployed(reason, message string) {
+// MarkDeployed registers the deployment completion time and sets the AllComponentsDeployed status in the
+// Release to True with the provided reason and message.
+func (r *Release) MarkDeployed(reason, message string, deployedTime metav1.Time) {
```

Comment:
Don’t pass the time. Same as the existing functions for the release. The moment you call it, the time is generated internally. We want the completion time to our eyes, not the real one.

Reply:
But then if the operation is called again, we will be updating the deployed time to right now when there could be no change at all in the deployment.. something else could have caused the release to be reconciled again

Reply:
Same in the release. We have guards for that.

Reply:
fixed